sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -408,6 +416,35 @@ public String getId() {
         return queryItemsInternal(new SqlQuerySpec(query), new CosmosQueryRequestOptions(), classType);
     }
 
+    /**
+     * Initializes the container by warming up the caches and connections.
+     * @return Mono of Void
+     */
+    @Beta(value = Beta.SinceVersion.V4_13_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public Mono<Void> initializeContainerAsync() {
+        if(this.getDatabase().getClient().getDocClientWrapper().getConnectionPolicy().getConnectionMode().equals(ConnectionMode.GATEWAY)) {
+            logger.info("Client is created with gateway mode, initialization of container is not needed, no action triggered");
```

Comment:
even in GW mode many caches need to be populated which are needed for reducing query execution time later.

CollectionCache
PartitionKeyRangeCache

Reply:
Done